be/src/exec/pipeline/pipeline_driver_executor.cpp

```diff
@@ -445,8 +449,7 @@ RuntimeProfile* GlobalDriverExecutor::_build_merged_instance_profile(QueryContex
             continue;
         }
 
-        auto* merged_driver_profile =
-                RuntimeProfile::merge_isomorphic_profiles(query_ctx->object_pool(), driver_profiles);
+        auto* merged_driver_profile = RuntimeProfile::merge_isomorphic_profiles(obj_pool, driver_profiles);
```

Comment:
This function (merge_isomorphic_profiles) should be changed too no need to add all profile in the obj

Reply:
This method will be invoked recursively, if not using ObjectPool, maybe you need a std::vector<std::shared_ptr<RuntimeProfile>> to collect all new-created profile objects, which is similar to using ObjectPool directly.